ibc-core/ics02-client/types/src/msgs/recover_client.rs

```diff
@@ -45,11 +46,11 @@ impl TryFrom<RawMsgRecoverClient> for MsgRecoverClient {
             subject_client_id: raw
                 .subject_client_id
                 .parse()
-                .map_err(ClientError::InvalidClientIdentifier)?,
+                .map_err(DecodingError::Identifier)?,
             substitute_client_id: raw
                 .substitute_client_id
                 .parse()
-                .map_err(ClientError::InvalidClientIdentifier)?,
+                .map_err(DecodingError::Identifier)?,
```

Comment:
It would be good to add `impl From<IdentifierError> for ClientError`, so we won’t need to use `map_err` in places like this.

Reply:
[e204dbc](https://github.com/cosmos/ibc-rs/pull/1325/commits/e204dbc3af6fed93fb57d3625290f5b525d9cb0b)